auth/rest.go

```diff
@@ -114,6 +121,82 @@ func (a *RestAuthDecorator) DecorateHandler(orig httprouter.Handle, appName stri
 			}
 
 			orig(res, req, p)
+
+			// if app was a provider app allow token rewrites
+			if cfg.Authentication.ProviderConfig.Service == appName {
```

Comment:
this should most likely not be in here i guess

Reply:
Maybe at least extract it into its own function?